### PR TITLE
Improve error message when missing the package-root

### PR DESCRIPTION
**Pull request by abarth**
@devoncarew @apwilson

### CODE REVIEW

File: lib/src/commands/flutter_command_runner.dart [hunk 1]
```diff
@@ -88,6 +88,18 @@ class FlutterCommandRunner extends CommandRunner {
       Logger.root.level = Level.FINE;
 
     ArtifactStore.packageRoot = globalResults['package-root'];
+    if (!FileSystemEntity.isDirectorySync(ArtifactStore.packageRoot)) {
+      String message = '${ArtifactStore.packageRoot} is not a valid directory.';
+      if (ArtifactStore.packageRoot == 'packages') {
+        if (FileSystemEntity.isFileSync('pubspec.yaml'))
+          message += '\nDid you run `pub get` in this directory?';
+        else
+          message += '\nDid you run this command from the same directory as your pubspec.yaml file?';
```
Comment (devoncarew) [inline lib/src/commands/flutter_command_runner.dart]: In some dart CLI tools I have put in a fast-fail at startup if the tool isn't run from a project root. May be something to consider here -

**Comment by devoncarew** [comment]
lgtm

File: lib/src/commands/flutter_command_runner.dart (hunk 1 above)
Comment (abarth) [inline lib/src/commands/flutter_command_runner.dart]: Is there a good pattern to follow for that?  Returning 2 below will exit the tool, but I'm not sure how best to detect whether we're at a project root.

Comment (devoncarew) [inline lib/src/commands/flutter_command_runner.dart]: A pubspec.yaml check is what I do. I'll open an issue to track this.

**Pull request merged**